packages/form/src/TextInputRow.js

```diff
@@ -95,55 +70,51 @@ class TextInputRow extends PureComponent {
         this.props.onBlur(event);
     }
 
-    handleInputChange = (event) => {
-        this.updateInputHeight(event.target);
-        this.props.onChange(event);
-    }
-
     renderInput({
         multiLine,
-        style: inputStyle = {},
-        ...inputProps
+        style = {},
+        renderInput,
+        inputComponent,
+        minRows,
+        maxRows,
+        ...otherProps
     }) {
         const sharedProps = {
-            ref: this.setInputRef,
             className: BEM.input.toString(),
+            style,
             onFocus: this.handleInputFocus,
             onBlur: this.handleInputBlur,
+            ...otherProps
         };
 
-        const textareaStyle = {
-            ...inputStyle,
-            height: this.state.inputHeight,
+        if (multiLine) {
+            const textareaProps = {
+                ...sharedProps,
+                minRows,
+                maxRows,
+            };
+
+            return <AutoSizeTextarea {...textareaProps} />;
+        }
+
+        const inputProps = {
+            type: 'text',
+            ...sharedProps,
         };
 
-        if (multiLine) {
-            return (
-                <textarea
-                    style={textareaStyle}
-                    onChange={this.handleInputChange}
-                    {...sharedProps}
-                    {...inputProps} />
-            );
+        if (isFunction(renderInput)) {
+            return renderInput(inputProps);
         }
 
-        return (
-            <input
-                type="text"
-                onChange={this.props.onChange}
-                {...sharedProps}
-                {...inputProps} />
+        return React.createElement(
+            isFunction(inputComponent) ? inputComponent : 'input',
```

Comment:
Maybe you can just set default value of `'input'` to `props.inputComponent`? And allow it to be either a String or a Function.

Reply:
Sounds great!